components/post_view/channel_intro_message/pluggable_intro_buttons/pluggable_intro_buttons.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+import {Channel} from '@mattermost/types/channels';
+
+import {PluginComponent} from 'types/store/plugins';
+
+type Props = {
+    channel: Channel;
+    pluginButtons: PluginComponent[];
+}
+
+const PluggableIntroButtons = React.memo((props: Props) => {
+    const intl = useIntl();
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    if (channelIsArchived || props.pluginButtons.length === 0) {
+        return null;
+    }
+
+    const buttons = props.pluginButtons.map((buttonProps) => {
+        // text defaults to "Create a board" because the original version of this API had hardcoded text and was
+        // built specifically for Boards
+        const text = buttonProps.text ?? intl.formatMessage({id: 'intro_messages.createBoard', defaultMessage: 'Create a board'});
```

Comment:
Boards doesn't pass in any text only "Boards" for a `tooltipText` option that's not used, so this is a bit of a hack to ensure that this is backwards compatible with the plugin. We can remove it once we stop supporting the Boards plugin